app/views/Project/Tagging/Sources/SourcesFilter/index.tsx

```diff
@@ -24,62 +26,274 @@ import {
     IoChevronUpOutline,
     IoChevronDownOutline,
 } from 'react-icons/io5';
-
-import { useRequest } from '#base/utils/restRequest';
-import { hasNoData } from '#utils/common';
+import { gql, useQuery } from '@apollo/client';
 import _ts from '#ts';
-import { KeyValueElement, LeadOptions, EmmEntity } from '#types';
+import { EnumFix } from '#utils/types';
+import {
+    hasNoData,
+    enumKeySelector,
+    enumLabelSelector,
+} from '#utils/common';
 import NonFieldError from '#components/NonFieldError';
-import { FilterFormType } from '../utils';
+import {
+    LeadEntriesFilterData,
+    ProjectSourcesQueryVariables,
+    SourceFilterOptionsQuery,
+    SourceFilterOptionsQueryVariables,
+} from '#generated/types';
 
+import { getValidDateRangeValues } from '../utils';
+import EntryFilter from './EntryFilter';
 import styles from './styles.css';
 
-// FIXME: Created at and published on are date ranges and not date inputs
+export type SourcesFilterFields = PurgeNull<Pick<EnumFix<ProjectSourcesQueryVariables,
+    'statuses'
+    | 'confidentiality'
+    | 'exists'
+    | 'priorities'
+    | 'statuses'
+>,
+    'statuses'
+    | 'search'
+    | 'exists'
+    | 'priorities'
+    | 'confidentiality'
+    | 'assignees'
+    | 'authoringOrganizationTypes'
+    | 'customFilters'
+>> & {
+    createdAt?: {
+        startDate: string;
+        endDate: string;
+    }
+    publishedOn?: {
+        startDate: string;
+        endDate: string;
+    }
```

Comment:
We shouldn't need these object values for createdAt

Reply:
@tnagorra right now we are using DateRangeInput and I believe we require it this way.

Reply:
We have already created simplified date range input.

Reply:
@tnagorra  you mean https://github.com/the-deep/deep-ui/tree/feature/date-range-dual-input ? This does not work as expected right now. it updates the parent form object instead of the property itself. for example 
expected value after onChange
`
{ 
    startDate: '2021-1-11',
    endDate: '2021-2-22', 
 }` 
 actual output
 `{ startDate: '2021-1-12' } { endDate: '2021-2-22' } `
 `